Pxb 8.0 2729 PXB compilation fails if GTest packages are installed

rahulmalik87 opened the pull request:
PXB-2729 PXB compilation fails if GTest packages are installed
    https://jira.percona.com/browse/PXB-2729

    Fixed the compilation error and xbcloud-t.cc test case

satya-bodapati reviewed (commented):
@rahulmalik87 can you please check why there are three commits? specially the "2740" commit id shouldn't have been present

rahulmalik87 commented:
> @rahulmalik87 can you please check why there are three commits? specially the "2740" commit id shouldn't have been present

Thank you @satya-bodapati. I was using wrong commit id of 2.4. Fixed my scripts.

satya-bodapati commented on storage/innobase/xtrabackup/src/xbcloud/xbcloud-t.cc:
```diff
@@ -489,3 +493,97 @@ TEST(keystone_v3, project_scoped_with_project_domain_keystone_v3_success) {
+          AllOf(IsPut(),
+                Property(
+                    &Http_request::url,
+                    StrEq("https://my-storage-account.blob.core.windows.netb/"
```
is the .netb a typo here? or expected?

rahulmalik87 replied:
b was for container , I have changed it to bucket.

satya-bodapati reviewed (changes requested):
Do these unit tests pass? can you please add jenkins result( assuming jenkins execute these unit tests)

satya-bodapati commented:
> Do these unit tests pass? can you please add jenkins result( assuming jenkins execute these unit tests)

@rahulmalik87 can you plz submit it to Jenkins?

Commit:
PXB-2729 PXB compilation fails if GTest packages are installed
https://jira.percona.com/browse/PXB-2729

Fixed the compilation error and xbcloud-t.cc test case

satya-bodapati reviewed (changes requested):
Please see above comments @rahulmalik87 ^

rahulmalik87 commented:
> > Do these unit tests pass? can you please add jenkins result( assuming jenkins execute these unit tests)
> 
> @rahulmalik87 can you plz submit it to Jenkins?

https://pxb.cd.percona.com/job/percona-xtrabackup-8.0-test-param/199. Test was running with -DWITH_UNIT_TESTS=ON. This unit test needs cmake version > 3.22.1 which is not installed on jenkin. so unit test are skipped. but it is succesfull on local.

satya-bodapati commented:
> > > Do these unit tests pass? can you please add jenkins result( assuming jenkins execute these unit tests)
> > 
> > 
> > @rahulmalik87 can you plz submit it to Jenkins?
> 
> https://pxb.cd.percona.com/job/percona-xtrabackup-8.0-test-param/199. Test was running with -DWITH_UNIT_TESTS=ON. This unit test needs cmake version > 3.22.1 which is not installed on jenkin. so unit test are skipped. but it is succesfull on local.

Where should I look that unitests are skipped, is it this?
main.unittests                           w6	[passed]    0

Can you please create a JEN issue?

@rahulmalik87 ^

satya-bodapati commented:
> > > > Do these unit tests pass? can you please add jenkins result( assuming jenkins execute these unit tests)
> > > 
> > > 
> > > @rahulmalik87 can you plz submit it to Jenkins?
> > 
> > 
> > https://pxb.cd.percona.com/job/percona-xtrabackup-8.0-test-param/199. Test was running with -DWITH_UNIT_TESTS=ON. This unit test needs cmake version > 3.22.1 which is not installed on jenkin. so unit test are skipped. but it is succesfull on local.
> 
> Where should I look that unitests are skipped, is it this? main.unittests w6 [passed] 0
> 
> Can you please create a JEN issue?
> 
> @rahulmalik87 ^

@rahulmalik87 please reply to this question. Can you please add proof that the new/modified unittests are passing?

rahulmalik87 commented:
> > > > > Do these unit tests pass? can you please add jenkins result( assuming jenkins execute these unit tests)
> > > > 
> > > > 
> > > > @rahulmalik87 can you plz submit it to Jenkins?
> > > 
> > > 
> > > https://pxb.cd.percona.com/job/percona-xtrabackup-8.0-test-param/199. Test was running with -DWITH_UNIT_TESTS=ON. This unit test needs cmake version > 3.22.1 which is not installed on jenkin. so unit test are skipped. but it is succesfull on local.
> > 
> > 
> > Where should I look that unitests are skipped, is it this? main.unittests w6 [passed] 0
> > Can you please create a JEN issue?
> > @rahulmalik87 ^
> 
> @rahulmalik87 please reply to this question. Can you please add proof that the new/modified unittests are passing?

{qa-sm-twina02 x8 o8 test}$more results/unittests
/home/rahul.malik/MySQL/src/x8/bld/storage/innobase/xtrabackup/test/../src/xbcloud/xbcloud-t
2022-06-17 09:26:33: run.sh: ===> xbcloud-t
Running main() from /home/rahul.malik/MySQL/src/googletest/googletest/src/gtest_main.cc
[==========] Running 16 tests from 6 test suites.
[----------] Global test environment set-up.
[----------] 3 tests from s3_client
[ RUN      ] s3_client.basicDNSv4
220617 09:26:33 (null): Successfully connected.
[       OK ] s3_client.basicDNSv4 (0 ms)
[ RUN      ] s3_client.basicPATHv4
220617 09:26:33 (null): Successfully connected.
[       OK ] s3_client.basicPATHv4 (0 ms)
[ RUN      ] s3_client.basicEndpoint
220617 09:26:33 (null): Successfully connected.
[       OK ] s3_client.basicEndpoint (0 ms)
[----------] 3 tests from s3_client (1 ms total)

[----------] 4 tests from s3v4_signer
[ RUN      ] s3v4_signer.basicDNS
[       OK ] s3v4_signer.basicDNS (0 ms)
[ RUN      ] s3v4_signer.basicPATH
[       OK ] s3v4_signer.basicPATH (0 ms)
[ RUN      ] s3v4_signer.sessionToken
[       OK ] s3v4_signer.sessionToken (0 ms)
[ RUN      ] s3v4_signer.storageClass
[       OK ] s3v4_signer.storageClass (0 ms)
[----------] 4 tests from s3v4_signer (0 ms total)

[----------] 2 tests from s3v2_signer
[ RUN      ] s3v2_signer.basicDNS
[       OK ] s3v2_signer.basicDNS (0 ms)
[ RUN      ] s3v2_signer.basicPATH
[       OK ] s3v2_signer.basicPATH (0 ms)
[----------] 2 tests from s3v2_signer (0 ms total)

[----------] 3 tests from keystone_v3
[ RUN      ] keystone_v3.unscoped_keystone_v3_success
[       OK ] keystone_v3.unscoped_keystone_v3_success (0 ms)
[ RUN      ] keystone_v3.project_scoped_keystone_v3_success
[       OK ] keystone_v3.project_scoped_keystone_v3_success (0 ms)
[ RUN      ] keystone_v3.project_scoped_with_project_domain_keystone_v3_success
[       OK ] keystone_v3.project_scoped_with_project_domain_keystone_v3_success (0 ms)
[----------] 3 tests from keystone_v3 (0 ms total)

[----------] 2 tests from azure_client
[ RUN      ] azure_client.basicDNSv4
[       OK ] azure_client.basicDNSv4 (0 ms)
[ RUN      ] azure_client.basicEndpoint
[       OK ] azure_client.basicEndpoint (0 ms)
[----------] 2 tests from azure_client (0 ms total)

[----------] 2 tests from azure_signer
[ RUN      ] azure_signer.basicDNS
[       OK ] azure_signer.basicDNS (0 ms)
[ RUN      ] azure_signer.storageClass
[       OK ] azure_signer.storageClass (0 ms)
[----------] 2 tests from azure_signer (0 ms total)

[----------] Global test environment tear-down
[==========] 16 tests from 6 test suites ran. (2 ms total)
[  PASSED  ] 16 tests.

Commit:
Merge branch 'PXB-2.4-2729' into pxb-8.0-2729
